contracts/FlashMintLib.sol

```diff
@@ -0,0 +1,154 @@
+	using SafeMath for uint256;
+	event Leverage(uint256 amountRequested, uint256 amountGiven, bool deficit, address flashLoan);
+
+	uint256 private constant PRICE_DECIMALS = 1e6;
+	uint256 private constant DAI_DECIMALS = 1e18;
+	uint256 private constant COLLAT_RATIO_DAI = 0.74 ether;
+    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address public constant CDAI = 0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643;
+	address private constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+	address private constant WBTC = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+	ComptrollerI private constant COMP = ComptrollerI(0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B);
+    address public constant LENDER = 0x1EB4CF3A948E7D72A198fe073cCb8C7a948cD853;
+    bytes32 public constant CALLBACK_SUCCESS = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    function doFlashMint(bool deficit, uint256 amountDesired, address want, uint256 maxFee) public returns (uint256) {
+        if(amountDesired == 0) {
+            return 0;
+        }
+
+        // calculate amount of ETH we need
+        (uint256 requiredDAI, uint256 amountWant) = getFlashLoanParams(want, amountDesired);
+
+        bytes memory data = abi.encode(deficit, amountWant);
+        uint256 _allowance = IERC20(DAI).allowance(address(this), address(LENDER));
+        uint256 _fee = IERC3156FlashLender(LENDER).flashFee(DAI, amountWant);
+        // Check that fees have not been increased without us knowing
+        require(_fee <= _calcFee(requiredDAI, maxFee));
+        uint256 _repayment = requiredDAI.add(_fee);
+        if(_allowance < _repayment) {
+            IERC20(DAI).approve(address(LENDER), 0);
+            IERC20(DAI).approve(address(LENDER), type(uint256).max);
+        }
+        IERC3156FlashLender(LENDER).flashLoan(IERC3156FlashBorrower(address(this)), DAI, requiredDAI, data);
+
+        emit Leverage(amountDesired, requiredDAI, deficit, address(LENDER));
+
+        return amountWant;
+    }
+
+    function maxLiquidity() public view returns (uint256) {
+        return IERC3156FlashLender(LENDER).maxFlashLoan(DAI);
+    }
+
+    function _calcFee(uint256 amount, uint256 maxFee) internal view returns (uint256) {
+        return amount.mul(maxFee).div(1e18);
+    }
+
+	function getFlashLoanParams(address want, uint256 amountDesired) internal returns (uint256 requiredDAI, uint256 amountWant) {
+		(uint256 priceDAIWant, uint256 decimalsDifference, uint256 _requiredDAI) = getPriceDAIWant(want, amountDesired);
+		// to avoid stack too deep	
+		requiredDAI = _requiredDAI;
+		amountWant = amountDesired;
+		// Not enough want in DAI. So we take all we can
+		uint256 _maxFlashLoan = maxLiquidity();
+        if(requiredDAI > _maxFlashLoan) {
+            requiredDAI = _maxFlashLoan.mul(95).div(100); // use only 95% of total liquidity
+            amountWant = requiredDAI.mul(COLLAT_RATIO_DAI).mul(PRICE_DECIMALS).div(priceDAIWant).div(1e18).div(decimalsDifference);
```

Comment:
if want is dai we only get 74% of the amountWant that we could take. if want is dai amountWant = requiredDAI

Reply:
div(decimalsDifference) this only works if want has <= decimals to dai. if it has more it breaks. We can probably ignore this situation but you do include it in getPriceDAIWant